default/python/AI/FreeOrionAI.py

```diff
@@ -343,7 +347,6 @@ def generateOrders():  # pylint: disable=invalid-name
     debug("Calling AI Modules")
     # call AI modules
     action_list = [
-        ColonisationAI.survey_universe,
```

Comment:
Maybe just move `aistate.prepare_for_new_turn()` into the action list after `ColonisationAI.survey_universe`?  

Most of the code before `action_list` is timers and debug prints, so I'd prefer survey universe calls after that.


Actually `prepare_for_new_turn()` is a number of small functions without arguments,  so we could just "inline" it here and put all calls to the list directly.   This will give us the flexibility to adjust orders.

The survey universe does a single cycle over everything and collects information.   
 Algorithmically doing this cycle twice does not look good.  But this could simplify the logic.   If we will have 2 or more cycles we could deal with temporal coupling in much easy way.

Reply:
survey_universe must be called before this [line](https://github.com/freeorion/freeorion/blob/df647152ccc177103d82d64e4c19ed9e7591a998/default/python/AI/FreeOrionAI.py#L312) because get_capital needs the information.

I wonder whether we can get rid of survey_universe completely by doing the calculation in the cache_for_current_turn-functions, like it did it for _get_pilot_summary (see second commit). We wouldn't even need the lock anymore then, but of course we would have a dozen loops instead of one.